timeserio/preprocessing/datetime.py

```diff
@@ -284,16 +315,76 @@ def transform(self, df):
-        return set(columns)
 
-    def transformed_columns(self, input_columns):
-        input_columns = set(_as_list_of_str(input_columns))
-        if not self.required_columns <= input_columns:
-            raise ValueError(f'Required columns are {self.required_columns}')
-        lags = _as_list_of_str(self.lags)
-        columns = _as_list_of_str(self.columns)
-        new_columns = [f"{col}_{lag}" for lag in lags for col in columns]
-        return input_columns | set(new_columns)
+class RollingMeanFeaturizer(_BaseLagFeaturizer):
+    """Add rolling mean features to a DataFrame.
+
+        Note that None will be inserted where no historic data is available.
+
+    Parameters:
+        datetime_column: str
+        columns: str or List[str]
+            Feature column to lag according to datetime_column.
+            See pandas.DataFrame.groupby
+        windows: List[int or offset]
+            List of moving window sizes.
+            See pandas.DataFrame.rolling
+        min_periods: int, default ``None``
+            Minimum number of observations in window required to have a value
+        center: bool, default False
+            Set the labels at the center of the window
+        win_type: str, default ``None``
+            Provide a window type. If ``None``, all points are
+            evenly weighted
+        closed: str, default ``None``
+            Make the interval closed on the 'right', 'left', 'both'
+            or 'neither' endpoints.
+        refit: bool, default True
+            If set to False, a fitted LagFeaturizer will not be refitted
+            on subsequent calls to `fit()`
+        duplicate_agg: str, default 'raise'
+            Aggregation functions to apply to values for duplicate datetimes.
+            By default, an error is raised if duplicates
+            See pandas.DataFrame.groupby().aggregate()
+
+    This Transformer is stateful.
+
+    """
+
+    def __init__(
+        self,
+        datetime_column: str,
+        columns: List[str],
+        windows: List,
+        min_periods: int = None,
+        center: bool = False,
+        win_type: str = None,
+        closed: str = None,
+        refit: bool = True,
+        duplicate_agg: str = 'raise'
+    ):
+        super().__init__(datetime_column, columns, refit, duplicate_agg)
+        self._lags = windows
+        self.min_periods = min_periods
+        self.center = center
+        self.win_type = win_type
+        self.closed = closed
+
+    def transform(self, df):
+        check_is_fitted(self, 'df_')
+        windows = _as_list_of_str(self._lags)
+        df = df.copy().set_index(self._datetime_column)
+        for window in windows:
+            window_df = self.df_.rolling(
+                window=window,
+                min_periods=self.min_periods,
+                center=self.center,
+                win_type=self.win_type,
+                closed=self.closed
+            ).mean().add_suffix(f"_{window}")
```

Comment:
This will result in names compatible with the `transformed_columns`  above :heavy_check_mark:
Sinve the `for window in windows`/ `for lag in lags` logic is shared, would it make sense to split out the method that computes `lag_df`/`window_df`? This would be the only method that needs subclassing.